src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php

```diff
@@ -54,10 +54,8 @@ public function __construct(TokenStorageInterface $tokenStorage, EventDispatcher
      *
      * @param string $providerKey The name of the provider/firewall being used for authentication
      */
-    public function authenticateWithToken(TokenInterface $token, Request $request/*, string $providerKey */)
+    public function authenticateWithToken(TokenInterface $token, Request $request, string $providerKey)
```

Comment:
hum actually we should add null as a default value here, so not break BC at least (and thus revert the changes on the test cases)

Reply:
No problem to make this change, I just observed in the project Symfony that all the features using this function implement the providerKey argument and as the class is annotate final I believed it will be OK, only the unit tests was needed to be changed

Reply:
if anyone else calls the function, their code will break. since the method is not internal, that's should remain supported.